mongodbatlas/fw_resource_mongodbatlas_search_deployment_model_test.go

```diff
@@ -0,0 +1,108 @@
+package mongodbatlas
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"go.mongodb.org/atlas-sdk/v20231115001/admin"
+)
+
+type sdkToTFModelTestCase struct {
+	SDKResp         *admin.ApiSearchDeploymentResponse
+	expectedTFModel *tfSearchDeploymentRSModel
+	name            string
+	clusterName     string
+}
+
+func TestSearchDeploymentSDKToTFModel(t *testing.T) {
+	testCases := []sdkToTFModelTestCase{
+		{
+			name:        "Complete SDK response",
+			clusterName: "Cluster0",
+			SDKResp: &admin.ApiSearchDeploymentResponse{
+				Id:        admin.PtrString("111111111111111111111111"),
+				GroupId:   admin.PtrString("222222222222222222222222"),
+				StateName: admin.PtrString("IDLE"),
+				Specs: []admin.ApiSearchDeploymentSpec{
+					{
+						InstanceSize: "S20_HIGHCPU_NVME",
+						NodeCount:    2,
+					},
+				},
+			},
+			expectedTFModel: &tfSearchDeploymentRSModel{
+				ID:          types.StringValue("111111111111111111111111"),
+				ClusterName: types.StringValue("Cluster0"),
+				ProjectID:   types.StringValue("222222222222222222222222"),
+				StateName:   types.StringValue("IDLE"),
+				Specs:       tfSpecsList(t, "S20_HIGHCPU_NVME", 2),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel, diags := newTFSearchDeployment(context.Background(), tc.clusterName, tc.SDKResp, nil)
+			if diags.HasError() {
+				t.Errorf("Unexpected errors found")
+			}
+			if !reflect.DeepEqual(resultModel, tc.expectedTFModel) {
+				t.Errorf("created terraform model did not match expected output")
+			}
+		})
+	}
+}
+
+type tfToSDKModelTestCase struct {
+	name           string
+	tfModel        *tfSearchDeploymentRSModel
+	expectedSDKReq admin.ApiSearchDeploymentRequest
+}
+
+func TestSearchDeploymentTFModelToSDK(t *testing.T) {
+	testCases := []tfToSDKModelTestCase{
```

Comment:
worth it adding other test cases with not all fields?

Reply:
In the case of search deployment there are no optional parameters that can be provided. Everything is required, as well as spec lists which has to be of 1 element (validated by Atlas API).